frontend/src/components/Forms/reports/ReportFilterForm.js

```diff
@@ -0,0 +1,216 @@
+    dropdownMineReportCategoryOptions,
+    selectedMineReportDefinitionGuid
+  ) => {
+    let dropdownMineReportCategoryOptionsFiltered = dropdownMineReportCategoryOptions;
+
+    if (selectedMineReportDefinitionGuid) {
+      const selectedMineReportDefinition = this.props.mineReportDefinitionOptions.filter(
+        (option) => option.mine_report_definition_guid === selectedMineReportDefinitionGuid
+      )[0];
+
+      dropdownMineReportCategoryOptionsFiltered = dropdownMineReportCategoryOptions.filter((cat) =>
+        selectedMineReportDefinition.categories
+          .map((category) => category.mine_report_category)
+          .includes(cat.value)
+      );
+    }
+    this.setState({
+      dropdownMineReportCategoryOptionsFiltered,
+    });
+  };
+
+  componentWillReceiveProps = (nextProps) => {
+    if (nextProps.selectedMineReportCategory !== this.props.selectedMineReportCategory) {
+      this.updateMineReportOptions(
+        nextProps.mineReportDefinitionOptions,
+        nextProps.selectedMineReportCategory
+      );
+    }
+    if (
+      nextProps.selectedMineReportDefinitionGuid !== this.props.selectedMineReportDefinitionGuid
+    ) {
+      this.updateMineReportCategoryOptions(
+        nextProps.dropdownMineReportCategoryOptions,
+        nextProps.selectedMineReportDefinitionGuid
+      );
+    }
+  };
+
+  render() {
+    return (
+      <Form
+        layout="vertical"
+        onSubmit={this.props.handleSubmit}
+        onReset={this.handleReset}
+        enableReinitialize="true"
+      >
+        <div>
+          <Row gutter={16}>
+            <Col md={6} sm={24}>
+              <Field
+                id="report_name"
+                name="report_name"
+                label="Report Name"
+                placeholder="Start typing a report name"
+                component={renderConfig.SELECT}
+                data={this.state.dropdownMineReportDefinitionOptionsFiltered}
+              />
+            </Col>
+            <Col md={6} sm={24}>
+              <Field
+                id="report_type"
+                name="report_type"
+                label="Report Type"
+                placeholder="Select a report type"
+                component={renderConfig.SELECT}
+                data={this.state.dropdownMineReportCategoryOptionsFiltered}
+              />
+            </Col>
+            <Col md={6} sm={24}>
+              <Field
+                id="compliance_year"
+                name="compliance_year"
+                label="Compliance Year"
+                placeholder="Start date"
+                component={renderConfig.YEAR}
+              />
+            </Col>
+            {/* <Col md={6} sm={24}>
+              <Field
+                id="requested_by"
+                name="requested_by"
+                label="Requested by"
+                placeholder="Start typing an inspector name"
+                component={renderConfig.SELECT}
+              />
+            </Col> */}
+          </Row>
```

Comment:
can this be removed?

Reply:
They can be but I left them in because they are meant for the non MVP version of Code required reports where more fields will be available.

Reply:
Okay, Either add a comment to explain the use for potential new developers or delete it

Reply:
Done.